### ISSUE TITLE
Steam Beta - Client inner window turns black on resize

### ISSUE THREAD

**Issue by DonKatsu** [ValveSoftware/steam-for-linux#9375]
#### Your system information

* Steam client version (build number or date): 1682634349
* Distribution (e.g. Ubuntu): Fedora 37 KDE (Wayland)
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Resizing or minimizing and restoring the Steam client causes the inside of the window to turn black. [Example screenshot](https://user-images.githubusercontent.com/8627704/235036568-08f9975b-d618-4174-b0e5-d0dce2c1dd34.png).
Navigating to other pages continues to show the black inner window. The only way I've found to fix it is to open the library page.
Resizing or minimizing and restoring Steam while in the Library doesn't cause it to happen.

It happens whether "GPU accelerated rendering in web views" is on or off.

#### Steps for reproducing this issue:

1. Resize or minimize Steam client window
2. Try navigating to a different page
3. Open library to fix

**Comment by KibaGasteiz** [comment]
Some more info here:
https://github.com/ValveSoftware/steam-for-linux/issues/9386

The issue only happens if you maximize moving the window to the top of the screen (or use Win+up on the keyboard), or if you resize it moving from top to middle of the screen (Win+down). So you can use the maximize button at the top right with as a workaround.

And yes, the black screen get fixed when going to the library section.

**Comment by z0rti** [comment]
The same still happens here with Steam-Version:  1686880776

**Comment by r2rX** [comment]
Can also confirm that this still occurs with Steam v1688792985. (Running Fedora 38 KDE Wayland w/ Mesa 23.2.0-git).

**Comment by LordJABA** [comment]
Switched to beta to check but in ver 1690426635 it is still happening - worse the same happens when starting games (ex. cyberpunk - same black square filling the game window). Started after switching from nvidia 4.XX to 5.XX drivers - currently on 535.86.05

**Comment by DonKatsu** [comment]
I'm not sure if it's from this issue or a new one, but  since a few beta updates ago my Library tab is acting up.

My Steam is set to open to the Library tab on launch. It behaves properly when Steam is initially launched.
When opening Steam to the Library tab through the notification panel icon from a closed window, it renders normally but appears unresponsive. It shows what the Library tab was last opened to, but it's still loading the default library page.
The cursor changes as I hover over clickable elements and text. If I change what game's info is shown by clicking one in the list of the Library page, the visual of the page doesn't change but the contents do.

This applies to UI elements as well such as the status indication color of my username next to the window controls.
If I launch a game from the notification panel icon while the window is in that state, it stays blue instead of changing to green.

Resizing the window updates what's shown and afterwards continues to properly render as long as the Steam window remains open. Changing tabs works when the Library is in its stuck state, but going back to the Library tab doesn't fix it.

The other options such as Store or Friend Activity show the same stuck UI behaviors, but renders a black page.
If I go from being on the Library tab, close the window, and open the Store from the panel icon, Library is underlined and in blue.

**Comment by DonKatsu** [comment]
Behavior update:
Dragging the Steam window from a maximized state into a restored state still causes the contents of the window to turn black.

Clicking the restore button doesn't cause the contents of the window to turn black, and the window can be resized without issue.
Clicking the maximize button then is fine, but dragging the Steam window to the top of the screen to maximize it causes the contents of the window to turn black.

Minimizing and restoring Steam still seems to sometimes cause the contents of the window to turn black, but not always. Not sure exactly.

**Comment by DonKatsu** [comment]
The April 26th Beta update (1714160074) does seem to have fixed this issue for me. On Plasma 6.
```
Linux
Fixed a case where the store tab would fail to display any content after maximizing the steam window.
```

**Comment by pm4rcin** [comment]
> The April 26th Beta update (1714160074) does seem to have fixed this issue for me. On Plasma 6.
> 
> ```
> Linux
> Fixed a case where the store tab would fail to display any content after maximizing the steam window.
> ```

Can confirm on Hyprland. Good to see it finally fixed :)

**Comment by scelusnet** [comment]
> The April 26th Beta update (1714160074) does seem to have fixed this issue for me. On Plasma 6.
> 
> ```
> Linux
> Fixed a case where the store tab would fail to display any content after maximizing the steam window.
> ```

Yep can confirm fedora.

**Comment by DonKatsu** [comment]
Actually, I have ran into at least one edge case that still ends up with a black window.
In Zomboid (but maybe any game with workshop integration), opening a workshop link in the in-game mod options redirects it to open the workshop page in the Steam client window.
When it does so, Steam ends up with the black page everywhere until fixing it by going to the Library tab again.

**Comment by manolol1** [comment]
Also still having that issue on Fedora 40 KDE (Plasma 6) with an Nvidia GeForce GTX 1060.

**Comment by kisak-valve** [comment]
Hello @DonKatsu, please open a new issue report focusing on your follow up issue.

Closing as fixed in the [2024-05-07](https://store.steampowered.com/news/app/593110/view/4182231197207095612) Steam client update.

**Comment by ruinedraziel** [comment]
Issue still occurs, but I posted in another thread, gonna repost it here. May want to merge this to issues.

https://github.com/ValveSoftware/steam-for-linux/issues/6660#issuecomment-2174352030

![image](https://github.com/ValveSoftware/steam-for-linux/assets/125985412/22820bfb-fa0b-4dbf-957f-7732ca72816c)

Same issue here,, but I have a very simple workaround, hope this helps.

    System: https://gist.github.com/ruinedraziel/9bba33b331d036d1538fe86ad6ff76fb
    I3-WM on Arch Linux

I can simply cause the issue at will by simply changing the workspace, if the workspace is with steam opened on background when I switch back to the workspace where steam is, the current page stop rendering, it seems to keep working, cause when the workaround is applied whenever I clicked for testing is where i'm at.

Simply pressing my full-screen I3 hotkey and back to normal mode mode fix the issue.
